server/nuxeo-launcher/src/main/java/org/nuxeo/launcher/process/UnixProcessManager.java

```diff
@@ -32,47 +33,52 @@
  * {@link ProcessManager} implementation for *nix systems. Uses the <tt>ps</tt> and <tt>kill</tt> commands.
  * <p>
  * Works for Linux. Works for Solaris too, except that the command line string returned by <tt>ps</tt> there is limited
- * to 80 characters and this affects {@link #findPid(String)}.
+ * to 80 characters and this affects {@link #findPid()}.
  */
-public class UnixProcessManager implements ProcessManager {
-
-    protected static final boolean PID_ENABLED = true;
+public class UnixProcessManager extends ProcessManager {
 
     private static final Pattern PS_OUTPUT_LINE = Pattern.compile("^\\s*(\\d+)\\s+(.*)$");
 
+    protected UnixProcessManager(Pattern processPattern) {
+        super(processPattern);
+    }
+
     protected String[] psCommand() {
         return new String[] { "/bin/ps", "-e", "-o", "pid,args" };
     }
 
     @Override
-    public String findPid(String regex) throws IOException {
-        Pattern commandPattern = Pattern.compile(regex);
-        for (String line : execute(psCommand())) {
-            Matcher lineMatcher = PS_OUTPUT_LINE.matcher(line);
+    public Optional<String> findPid() throws IOException {
+        for (String line : execute0(psCommand())) {
+            Matcher lineMatcher = lineMatcher(line);
             if (lineMatcher.matches()) {
                 String command = lineMatcher.group(2);
                 Matcher commandMatcher = commandPattern.matcher(command);
                 if (commandMatcher.find()) {
-                    return lineMatcher.group(1);
+                    return Optional.ofNullable(lineMatcher.group(1));
                 }
             }
         }
-        return null;
+        return super.findPid();
+    }
+
+    protected Matcher lineMatcher(String line) {
+        return PS_OUTPUT_LINE.matcher(line);
     }
 
     @Override
-    public void kill(Process process, String pid) throws IOException {
-        execute("/bin/kill", "-KILL", pid);
+    public void kill(ProcessHandle processHandle) throws IOException {
+        execute0("/bin/kill", "-KILL", String.valueOf(processHandle.pid()));
     }
 
-    protected List<String> execute(String... command) throws IOException {
-        Process process = new ProcessBuilder(command).start();
-        return IOUtils.readLines(process.getInputStream(), UTF_8);
+    // non-static method to allow tests to mock it
+    protected List<String> execute0(String... command) throws IOException {
```

Comment:
```suggestion
    protected List<String> execute0(String... commands) throws IOException {
```

Reply:
`command` is not plural because we aligned the naming on `ProcessBuilder`.
`command[0]` represents the unix/windows command, followed by its arguments, thus the singular I think.